doc/anaconda.md

```diff
@@ -0,0 +1,70 @@
+Cockpit Storage in Anaconda Mode
+================================
+
+Anaconda (the OS Installer) can open the Cockpit "storaged" page for
+advanced setup of the target storage devices. When this is done,
+storaged is in a special "Anaconda mode" and behaves significantly
+different.
+
+In essence, the storaged page restricts itself to working with the
+target environment. It will hide the real root filesystem (on the USB
+stick that the Live environment was booted from, say), but let the
+user create a "fake" root filesystem on some block device.
+
+Entering Anaconda mode
+----------------------
+
+The "storaged" page is put into Anaconda mode by storing a
+"cockpit_anaconda" item in its `window.localStorage`.  The value
+should be a JSON encoded object, the details of which are explained
+below.
+
+Since both Anaconda and the storaged page are served from the same
+origin, Anaconda can just execute something like this:
+
+```
+    window.localStorage.setItem("cockpit_anaconda",
+                                JSON.stringify({
+                                    "mount_point_prefix": "/sysroot",
+                                    "available_devices": [ "/dev/sda" ]
+                                }));
+    window.open("/cockpit/@localhost/storage/index.html", "storage-tab");
+```
+
+Ignoring storage devices
+------------------------
+
+Anaconda needs to tell Cockpit which devices can not be used to
+install the OS on. This is done with the "available_devices" entry,
+which is an array of strings.
+
+```
+{
+  "available_devices": [ "/dev/sda" ]
+}
+```
+
+This list should only contain entries for top-level block devices. It
+should not contain things like partitions, device mapper devices, or
+mdraid devices.
+
+Mount point prefix
+------------------
+
+Cockpit can be put into a kind of "chroot" environment by giving it a
+mount point prefix like so:
+
+```
+{
+  "mount_point_prefix": "/sysroot"
+}
+```
+
+This works at the UI level: filesystems that have mount points outside
+of "/sysroot" are hidden from the user, and when letting the user work
+with mount points below "/sysroot", the "/sysroot" prefix is
+omitted. So when the user says to create a filesystem on "/var", they
+are actually creating one on "/sysroot/var".
+
+However, Cockpit (via UDisks2) will still write the new mount point
+configuration into the real /etc/fstab (_not_ /sysroot/etc/fstab).
```

Comment:
Is that the way how C-Storage "hands over" back to Anaconda? I.e. Anaconda will inspect /etc/fstab to see what the user has done? Or does it even have to care? At first sight it seems Anaconda can just assume "whatever the user did to /sysroot, I'll just install into it", but then again (1) boot, EFI etc. partitions are special and possibly not even mounted, and (2) Anaconda will still have to write /sysroot/etc/fstab, so I suppose it needs to handle these somehow.

This paragraph could explain in what way the written /etc/fstab is relevant.

Reply:
> Is that the way how C-Storage "hands over" back to Anaconda?

No. We write to fstab because this is the easiest thing to do for Cockpit. I have explained that now.

Reply:
I see -- "If and how Cockpit communicates back to Anaconda is still open." . So I take it Anaconda doesn't actually care that much, it'll just happily copy stuff into /sysroot and not second-guess that? But how does it create /sysroot/etc/fstab then? Anyway, that issue is blindingly obvious once Anaconda starts to actually use this, so this can/will happen in a follow-up and doesn't need to block this PR. Thanks for making it explicit in the docs!